src/azul/service/manifest_service.py

```diff
@@ -407,43 +422,62 @@ def get_cached_manifest_with_object_key(self,
                                             object_key: str
                                             ) -> Optional[Manifest]:
         generator = ManifestGenerator.for_format(format_, self, catalog, filters)
-        presigned_url = self._get_cached_manifest(generator, object_key)
-        if presigned_url is None:
+        file_name = self._get_cached_manifest_file_name(generator, object_key)
+        if file_name is None:
             return None
         else:
+            presigned_url = self._presign_url(generator, object_key, file_name)
             return Manifest(location=presigned_url,
                             was_cached=True,
                             format_=format_,
                             catalog=catalog,
                             filters=filters,
-                            object_key=object_key)
-
-    def _get_cached_manifest(self,
-                             generator: 'ManifestGenerator',
-                             object_key: str
-                             ) -> Optional[str]:
-        if self._can_use_cached_manifest(object_key):
-            file_name = self._use_cached_manifest(generator, object_key)
-            return self._presign_url(object_key, file_name)
-        else:
-            return None
+                            object_key=object_key,
+                            file_name=file_name)
 
     file_name_tag = 'azul_file_name'
 
-    def _use_cached_manifest(self, generator: 'ManifestGenerator', object_key: str) -> Optional[str]:
+    def _get_cached_manifest_file_name(self,
+                                       generator: 'ManifestGenerator',
+                                       object_key: str
+                                       ) -> Optional[str]:
         """
-        Return the content disposition file name of the exiting cached manifest.
+        Return file_name of the manifest associated with the object_key if it
+        was previously created, still exists in the bucket, and won't be
+        expiring soon. Otherwise return None.
```

Comment:
```suggestion
        Return the proposed local file name of the manifest with the given object key if it
        was previously created, still exists in the bucket, and won't be
        expiring soon. Otherwise return None.
```